base/expr.jl

```diff
@@ -184,6 +184,56 @@ macro noinline(ex)
     esc(isa(ex, Expr) ? pushmeta!(ex, :noinline) : ex)
 end
 
+"""
+    @pure
+
+Usage can easily lead to whole program corruption or crashes and should be avoided
+by beginners.
+
+Do not use if the function involved:
+
+- Involves globals, pointers
+- Recurses
+- Does not return exactly (`===`) the same result for the same input
+- Gets its methods extended after it is called
+- Uses dispatch on one of its arguments
+
+### Example 
```

Comment:
This example isn't wrong, but it will almost certainly be made obsolete by #24362